### ISSUE TITLE
require-atomic-updates incorrectly considers out-of-scope variables

### ISSUE THREAD

**Issue by tkalliom** [eslint/eslint#14045]
**Tell us about your environment**

* **ESLint Version:** v7.18.0
* **Node Version:** v10.23.1
* **npm Version:** v6.14.8

**What parser (default, `@babel/eslint-parser`, `@typescript-eslint/parser`, etc.) are you using?**
default

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
module.exports = {
    "env": {
        "node": true,
        "es2021": true
    },
    "parserOptions": {
        "ecmaVersion": 12,
        "sourceType": "module"
    },
    "rules": {
        "require-atomic-updates": ["warn"]
    }
};

```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
async function getOtherList() {
    return Promise.resolve([{tag: "other"}]);
}

const main = async function () {
    const list = [{tag: "something"}];
    for (const item of list) {
        console.log(item.tag);
    }

    const otherList = await getOtherList();
    for (const item of list) {
        const match = otherList.find(o => o.tag === item.tag); // warning for this line
        console.log(match);
    }
}
```

```bash
npx eslint index.js
```

**What did you expect to happen?**
No warning to be produced, since there is no asynchronicity involved in the second loop.

**What actually happened? Please include the actual, raw output from ESLint.**
```
  12:5  warning  Possible race condition: `const item` might be reassigned based on an outdated value of `const item`  require-atomic-updates

✖ 1 problem (0 errors, 1 warning)
```

For comparison, the following code – with only the name of the iteration variable changed – produces no warnings.
```js
async function getOtherList() {
    return Promise.resolve([{tag: "other"}]);
}

const main = async function () {
    const list = [{tag: "something"}];
    for (const item of list) {
        console.log(item.tag);
    }

    const otherList = await getOtherList();
    for (const item2 of list) {
        const match = otherList.find(o => o.tag === item2.tag); // warning for this line
        console.log(match);
    }
}
```

**Are you willing to submit a pull request to fix this bug?**
no

**Comment by aladdin-add** [comment]
Duplicate of #11899
